packages/inference/src/lib/makeRequestOptions.ts

```diff
@@ -180,12 +212,19 @@ export function makeRequestOptionsFromResolvedModel(
 	const body = binary
 		? args.data
 		: JSON.stringify(
-				providerConfig.makeBody({
-					args: remainingArgs as Record<string, unknown>,
-					model: resolvedModel,
-					task,
-					chatCompletion,
-				})
+				input.provider === "hf-inference"
+					? providerConfigs[input.provider].makeBody({
+							args: remainingArgs as Record<string, unknown>,
+							model: input.resolvedModel,
+							task,
+							chatCompletion,
+					  })
+					: providerConfig.makeBody({
+							args: remainingArgs as Record<string, unknown>,
+							model: input.resolvedModel,
+							task,
+							chatCompletion,
+					  })
```

Comment:
Exact same code as before but needed to help TS compiler see that we use an `undefined`  model only with `hf-inference` provider